swift/ios-shoppe-demo/Views/CartTableViewCell.swift

```diff
@@ -12,14 +12,15 @@ import UIKit
 class CartTableViewCell: UITableViewCell {
     @IBOutlet weak var orderTotalLabel: UILabel!
 
-    var tableViewController: CartTableViewController?
+    weak var tableViewController: CartTableViewController?
 
     func setupCartOrder() {
         guard let cartOrderTotal = tableViewController?.order.cartOrderTotal() else {
             return
         }
 
-        orderTotalLabel.text = "$\(cartOrderTotal)"
+        orderTotalLabel.adjustsFontSizeToFitWidth = true
+        orderTotalLabel.text = "$\(abs(cartOrderTotal.distance(to: 0)))"
```

Comment:
What's the need for the intermediate calculation?  Is `cartOrderTotal` not valid as-is?

Reply:
No it has the tendency to be negative without that :( I need to do some more research on this